core/src/test/java/io/grpc/internal/ManagedChannelImplTest.java

```diff
@@ -1624,71 +1611,14 @@ public void resetConnectBackoff_noOpWhenNameResolverNotStarted() {
   }
 
   @Test
-  public void orphanedChannelsAreLogged() throws Exception {
-    int remaining = unterminatedChannels;
-    for (int retry = 0; retry < 3; retry++) {
-      System.gc();
-      System.runFinalization();
-      if ((remaining -= ManagedChannelReference.cleanQueue()) <= 0) {
-        break;
-      }
-      Thread.sleep(100L * (1L << retry));
-    }
-    assertThat(remaining).isAtMost(0);
-
-    createChannel(
-        new FakeNameResolverFactory(true),
-        NO_INTERCEPTOR,
-        false, // Don't create a transport, Helper maintains a ref to the channel.
-        ManagedChannelImpl.IDLE_TIMEOUT_MILLIS_DISABLE);
-    assertNotNull(channel);
-    LogId logId = channel.getLogId();
-
-    // Try to capture the log output but without causing terminal noise.  Adding the filter must
-    // be done before clearing the ref or else it might be missed.
-    final List<LogRecord> records = new ArrayList<LogRecord>(1);
-    Logger channelLogger = Logger.getLogger(ManagedChannelImpl.class.getName());
-    Filter oldFilter = channelLogger.getFilter();
-    channelLogger.setFilter(new Filter() {
-
-      @Override
-      public boolean isLoggable(LogRecord record) {
-        synchronized (records) {
-          records.add(record);
-        }
-        return false;
-      }
-    });
-
-    // TODO(carl-mastrangelo): consider using com.google.common.testing.GcFinalization instead.
-    try {
-      channel = null;
-      // That *should* have been the last reference.  Try to reclaim it.
-      boolean success = false;
-      for (int retry = 0; retry < 3; retry++) {
-        System.gc();
-        System.runFinalization();
-        int orphans = ManagedChannelReference.cleanQueue();
-        if (orphans == 1) {
-          success = true;
-          break;
-        }
-        assertEquals("unexpected extra orphans", 0, orphans);
-        Thread.sleep(100L * (1L << retry));
-      }
-      assertTrue("Channel was not garbage collected", success);
-
-      LogRecord lr;
-      synchronized (records) {
-        assertEquals(1, records.size());
-        lr = records.get(0);
-      }
-      assertThat(lr.getMessage()).contains("shutdown");
-      assertThat(lr.getParameters()).asList().containsExactly(logId, target).inOrder();
-      assertEquals(Level.SEVERE, lr.getLevel());
-    } finally {
-      channelLogger.setFilter(oldFilter);
-    }
+  public void orphanedRefClearedOnTermination() throws Exception {
+    FakeNameResolverFactory nameResolverFactory = new FakeNameResolverFactory(true);
+    createChannel(nameResolverFactory, NO_INTERCEPTOR, false /* requestConnection */,
+            ManagedChannelImpl.IDLE_TIMEOUT_MILLIS_DISABLE);
+    Reference<?> ref = mock(Reference.class);
```

Comment:
im slightly afraid of mocking special jvm classes like references.  consider extending it like normal and making sure clear is called.

Reply:
+1

Reply:
Removed this test